### DISCUSSION TITLE
Need better understanding on Vm.Logs

### DISCUSSION THREAD

**Discussion by Abhishek-2416** [Cyfrin/foundry-full-course-cu#1494] [Q&A]
```
Vm.Log[] memory entries = vm.getRecordedLogs();
        bytes32 requestId = entries[1].topics[1];
```
Need better understanding on the getting event info 
any specific resource i could refer except the foundry book

**Comment by EngrPips** [comment]
Hello @Abhishek-2416 , That functionality is something that is specific to foundry framework except if  you are talking about reading events data from the ethereum blockchain itself, for that i found this youtube video https://www.youtube.com/watch?v=lAw5GggnXxs , it seem like a good resources. i didn't take my time to watch it tho.

**Reply by Abhishek-2416** [reply]
This is for hardhat, was looking for something which i can use in the foundry tests

**Reply by EngrPips** [reply]
If you are looking for something that you can use in the foundry tests then i must confess i don't know a better resources than the documentation of foundry itself but then maybe this youtube link https://www.youtube.com/watch?v=tgs5q-GJmg4&list=PLO5VPQH6OWdUrKEWPF07CSuVm3T99DQki  would be helpful.

**Reply by Abhishek-2416** [reply]
Have watched smart contract programmer's content before he doesn't explain the reading of data from events

**Reply by EngrPips** [reply]
I am very sorry to inform you that i don't know any other resources to know and understand foundry `Vm.logs` except for foundry book itself.
